RDoc-2397 Document 6.0 docker Linux image breaking changes in readmes (#16082)
ravendb/ravendb: docker/readme.md

    - based on Windows Nanoserver image (run using Windows containers). 

> 

Reviewer: @reebhub We'll update our docker docs after this is merged. https://ravendb.net/docs/article-page/5.4/csharp/start/installation/running-in-docker-container
cc @ppekrol

### What's new

> RavenDB `v6.0` docker image includes **switching to a non-root user**, providing far better security.

Reviewer: ```suggestion
RavenDB `v6.0` docker image has been adjusted to run under **a non-root user**, providing far better security.
```

> Containers are now using RavenDB *.deb package*. More information about how to deal with the breaking changes in the **migration from 5.4 image** section.

Reviewer: Can you make a markdown link to this section here?

##### How to set a custom config file?

Mount it as a docker volume and use `--config-path PATH_TO_CONFIG` command line argument in order to use settings file from outside of server directory.

##### What migration process from 5.4 image looks like?

> Data is stored in a different directory, and needs to be migrated or linked on the update. The container is now using the .deb package, which is well documented [here](https://github.com/ravendb/ravendb/blob/v5.4/scripts/linux/pkg/linux-packaging.md). The link describes new filesystem locations and permissions. 

Reviewer: You might wanna use the docs link https://ravendb.net/docs/article-page/latest/csharp/start/installation/gnu-linux/deb

> Data is stored in a different directory, and needs to be migrated or linked on the update. The container is now using the .deb package, which is well documented [here](https://github.com/ravendb/ravendb/blob/v5.4/scripts/linux/pkg/linux-packaging.md). The link describes new filesystem locations and permissions. 

Reviewer: ```suggestion
Data is stored in a different directory, and needs to be migrated or linked upon update. The container is now using the .deb package, which is well documented [here](https://github.com/ravendb/ravendb/blob/v5.4/scripts/linux/pkg/linux-packaging.md). The link describes new filesystem locations and permissions. 
```

We highly recommend migrating your data from legacy data directory to the new one `/var/lib/ravendb/data/`.

> However, running the server using `run-server.sh`, which is an entrypoint for the container, causes to run `link-legacy-datadir.sh` script. It checks whether there is data stored in the legacy data directory. If so, it tries to create symlink with the new data directory. If the permissions are insufficent for the 'ravendb' user (which is running the container) it'll fail with appropriate error message.

Reviewer: ```suggestion
However, running the server using `run-server.sh`, which is a default entrypoint for the container, causes to run `link-legacy-datadir.sh` script. It checks whether RavenDB data is stored under legacy data directory. If so, it tries to create symlink to the new data directory. If the permissions are insufficent for the 'ravendb' user (which is running the container) it is going to fail with an appropriate error message.
```

##### How to run as different UID?

> To run with different UID or GID, you need to build the appropriate ubuntu image with these build args (both are optional):

Reviewer: ```suggestion
To run with a different UID or GID, you need to build the Ubuntu image yourself with these build args (both are optional):
```

` --build-arg "RAVEN_USER_ID=999" --build-arg "RAVEN_GROUP_ID=999"`

> The `ravendb` user will use the following UID/GID, it's being set in the .deb package post-installation process.

Reviewer: Please add information about the default UID and GID

##### How to set a custom config file?

Mount it as a docker volume and use `--config-path PATH_TO_CONFIG` command line argument in order to use settings file from outside of server directory.

> ##### What migration process from 5.4 image looks like?

Reviewer 2: where is the link to actually see this in the webstie?

Reviewer 3: PR is still opened: https://github.com/ravendb/docs/pull/1665

Data is stored in a different directory, and needs to be migrated or linked upon update. The container is now using the .deb package, which is well documented [here](https://ravendb.net/docs/article-page/latest/csharp/start/installation/gnu-linux/deb). The link describes new filesystem locations and permissions. 

We highly recommend migrating your data from legacy data directory to the new one `/var/lib/ravendb/data/`.

> However, running the server using `run-server.sh`, which is a default entrypoint for the container, causes to run `link-legacy-datadir.sh` script. It checks whether RavenDB data is stored under legacy data directory. If so, it tries to create symlink to the new data directory. If the permissions are insufficent for the 'ravendb' user (which is running the container) it is going to fail with an appropriate error message.

Reviewer 3: entrypoint => entry point

Author: done
